packages/dds/tree/src/simple-tree/api/treeNodeApi.ts

```diff
@@ -285,6 +250,70 @@ export function tryGetSchema(value: unknown): undefined | TreeNodeSchema {
 	}
 }
 
+/**
+ *
+ * Gets the identifier from a node.
+ *
+ * @param node - {@link TreeNode} where you want to extract the identifier from.
+ * @param isCompressed - boolean for whether or not to return the compressed identifier, if possible.
+ *
+ * @remarks
+ * If the node does not contain an identifier field, it returns undefined.
+ *
+ * If isCompressed is set to `true`:
+ * - If the node contains a compressible identifier known by the id compressor, the compressed identifier is returned.
+ * - If the node contains an identifier, but is not compressible or unknown by the id compressor, the uncompressed identifier is returned.
+ *
+ * If isCompressed is set to `false`:
+ * - If the node contains an identifier field, the uncompressed identifier is returned.
+ */
+export function getIdentifierFromNode(
+	node: TreeNode,
+	isCompressed: boolean,
```

Comment:
nit: The name `preferCompressed` would be less confusing to me. You might also consider adding overloads, which will make the contract of the return value more self-documenting:

```ts
function getIdentifierFromNode(node: TreeNode, preferCompressed: true): number | string | undefined;
function getIdentifierFromNode(node: TreeNode, preferCompressed: false): string | undefined;
```